File: FakturusTrack.Android/app/src/main/java/com/fakturus/track/services/subscription/FeatureGate.kt
package com.fakturus.track.services.subscription

import com.fakturus.track.R

enum class FeatureGate(
    val requiredTier: Tier,
    val displayNameRes: Int
) {
    // STARTER
    PDF_EXPORT(Tier.STARTER, R.string.feature_pdf_export),
    CSV_EXPORT(Tier.STARTER, R.string.feature_csv_export),
    SICK_DAYS(Tier.STARTER, R.string.feature_sick_days),
    VACATION(Tier.STARTER, R.string.feature_vacation),

    // PRO
    DATEV_EXPORT(Tier.PRO, R.string.feature_datev_export),
    SCHOOL_HOLIDAYS(Tier.PRO, R.string.feature_school_holidays),
    CALENDAR_INTEGRATION(Tier.PRO, R.string.feature_calendar_integration);
}

File: FakturusTrack.iOS/FakturusTrack/Services/Subscription/FeatureGate.swift
import Foundation

enum FeatureGate: CaseIterable, Sendable {
    // STARTER Features
    case pdfExport
    case csvExport
    case sickDays
    case vacation

    // PRO Features
    case datevExport
    case schoolHolidays
    case calendarIntegration

    var requiredTier: Tier {
        switch self {
        case .pdfExport, .csvExport,
             .sickDays, .vacation:
            return .starter
        case .datevExport, .schoolHolidays, .calendarIntegration:
            return .pro
        }
    }

    /// Lokalisierter Feature-Name fuer Paywall
    var displayName: String {
        switch self {
        case .pdfExport: String(localized: "feature_pdf_export")
        case .csvExport: String(localized: "feature_csv_export")
        case .sickDays: String(localized: "feature_sick_days")
        case .vacation: String(localized: "feature_vacation")
        case .datevExport: String(localized: "feature_datev_export")
        case .schoolHolidays: String(localized: "feature_school_holidays")
        case .calendarIntegration: String(localized: "feature_calendar_integration")
        }
    }
}
